### ISSUE TITLE
Rounding problems (never reaching 70% strength with gunpowder)

### ISSUE THREAD

**Issue by slayit** [McJty/DeepResonance#139]
When I try to push the RCL to 85/70/60 via gunpowder and purifiers it never reaches the 70 strength.
I've written a litte OpenComputers program to read and print out the values every second.

This is the setup: https://i.imgur.com/bO8zTZE.png https://i.imgur.com/dBp9JLd.png https://i.imgur.com/TCubRpS.jpg
Imagine purifiers at the four empty slots. I've removed them for testing in this screenshot.

This is the result with 4 purifiers and 2 lasers with gunpowder: http://i.giphy.com/3oz8xJ4dj1n13RfKFy.gif
This is the result with only the lasers and without the purifiers: http://i.giphy.com/l3vRojEsdOLgaGMOk.gif
As you can see without the purifiers running the efficiency goes slightly beyond 60%.

There is 52000mb in 5 tanks. The tooltip rounds the value up to 70%, but the valve never let's fluid through.
(Ignore that it's not 85% purity in the screenshot, that was only a result of me removing the purifiers for not wasting ressources and for testing)

deepresonance-1.10-1.3.1 (deepresonance-1.10-1.2.9 as well, have not tried 1.3.0)
ElecCore-1.10.2-1.6.345
mcjtylib-1.10-2.2.0

**Comment by McJty** [comment]
Yes, that's a known problem which I can't do a lot about given that this is all floating point

**Comment by slayit** [comment]
It is not a solution to go slightly above the threshold like with the purifiers? So gunpowder and similar things don't go to 70% but to 70.1%. (I think the purifiers goes 0.1 above its value)
Or only keep six places after the comma and then round?

**Comment by Saklad5** [comment]
Is there any reason you can’t use something like [BigDecimal](https://docs.oracle.com/javase/7/docs/api/java/math/BigDecimal.html) to effectively rule out issues like this? You can specify arbitrary precision and rounding behavior using [MathContext](https://docs.oracle.com/javase/7/docs/api/java/math/MathContext.html) too.
